core/pom.xml

```diff
@@ -41,9 +41,15 @@
         <dependency>
             <groupId>org.apache.hadoop</groupId>
             <artifactId>hadoop-minicluster</artifactId>
-            <version>2.7.4</version>
+            <version>${hadoop.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+          <groupId>org.mockito</groupId>
+          <artifactId>mockito-all</artifactId>
+          <version>1.8.5</version>
+          <scope>compile</scope>
```

Comment:
Why did this need to be added as an additional dependency? And if it is needed I assume this should be `test` scope

Reply:
> Why did this need to be added as an additional dependency?

Some versions of Hadoop 2.x (e.g., 2.10.2) doesn't have Mockito in the minicluster test-jar, and it leads to NoClassDefFoundError unless specifying this dependency explicitly. According to https://issues.apache.org/jira/browse/HADOOP-14857, it's supposed to be fixed in 3.x, but 3.2.4 still seems to have the same problem.